libs/langchain/langchain/agents/openai_functions_agent/agent_token_buffer_memory.py

```diff
@@ -0,0 +1,65 @@
+"""Memory used to save agent output AND intermediate steps."""
+from typing import Any, Dict, List
+
+from langchain.agents.openai_functions_agent.base import _format_intermediate_steps
```

Comment:
feels like this should be moved. we do have langchain.chains.openai_functions/utils, but at this pt maybe should just make a langchain.utilities.openai_functions

Reply:
probably. i didnt want to do that in pr but open to that in future